src/server/pegasus_mutation_duplicator.cpp

```diff
@@ -205,6 +230,75 @@ void pegasus_mutation_duplicator::on_duplicate_reply(uint64_t hash,
     }
 }
 
+void pegasus_mutation_duplicator::type_force_send_no_idempotent_if_need(duplicate_rpc &rpc)
+{
+
+    // there maybe more than one mutation in one dup rpc
+    if (FLAGS_force_send_no_idempotent_when_duplication) {
+        for (auto entry : rpc.request().entries) {
+            if (entry.task_code == dsn::apps::RPC_RRDB_RRDB_INCR ||
+                entry.task_code == dsn::apps::RPC_RRDB_RRDB_CHECK_AND_SET ||
+                entry.task_code == dsn::apps::RPC_RRDB_RRDB_CHECK_AND_MUTATE) {
+
```

Comment:
Add these codes in a set, then:

```
if (entry.task_code not in set) {
    continue;
}

METRIC_VAR_INCREMENT(dup_retry_no_idempotent_duplicate_qps);

switch (entry.task_code) {
case x:
...
case y:
...
}
```

Reply:
`entry.task_code` is not a const variable. And I think using task_code.to_string is not more efficient than `if` structure.